_docs/tasks/security/authn-policy.md

```diff
@@ -0,0 +1,300 @@
+
+```
+kubectl exec $(kubectl get pod -l app=sleep -n bar -o jsonpath={.items..metadata.name}) -c sleep -n bar -- curl http://httpbin.foo:8000/ip -s -o /dev/null -w "%{http_code}\n"
+```
+```
+200
+```
+
+Conveniently, this one-liner command iterates through all combinations:
+
+```
+for from in "foo" "bar" "legacy"; do for to in "foo" "bar"; do kubectl exec $(kubectl get pod -l app=sleep -n ${from} -o jsonpath={.items..metadata.name}) -c sleep -n ${from} -- curl http://httpbin.${to}:8000/ip -s -o /dev/null -w "sleep.${from} to httpbin.${to}: %{http_code}\n"; done; done
+```
+
+```
+sleep.foo to httpbin.foo: 200
+sleep.foo to httpbin.bar: 200
+sleep.bar to httpbin.foo: 200
+sleep.bar to httpbin.bar: 200
+sleep.legacy to httpbin.foo: 200
+sleep.legacy to httpbin.bar: 200
+```
+
+
+* Also verify that there are no authencation policy in the system
+
+```
+kubectl get policies.authentication.istio.io -n foo
+
+kubectl get policies.authentication.istio.io -n bar
+```
+
+```
+No resources found.
+```
+
+## Enable mTLS for all services in namespace `foo`.
+
+Run this command to set namespace-level policy for namespace `foo`.
+
+```
+cat <<EOF | istioctl create -f -
+apiVersion: "authentication.istio.io/v1alpha1"
+kind: "Policy"
+metadata:
+  name: "enable-mtls"
+  namespace: "foo"
+spec:
+  peers:
+  - mtls:
+EOF  
+```
+
+And verify policy is added:
+```
+kubectl get policies.authentication.istio.io -n foo
+```
+
+```
+NAME          AGE
+enable-mtls   1m
+```
+
+Run the same testing command above. We should see request from `sleep.legacy` to `httpbin.foo` start to fail, as the result of enabling mTLS for `httpbin.foo` but `sleep.legacy` doesn't have sidecar to support it. On the other hand, for clients with sidecar (`sleep.foo` and `sleep.bar`), Istio automatically configures them to using mTLS where talking `http.foo`, so they continue to work. Also, requests to `httpbin.bar` are not affected as policy is effective on `foo` namespace only.
+
+```
+for from in "foo" "bar" "legacy"; do for to in "foo" "bar"; do kubectl exec $(kubectl get pod -l app=sleep -n ${from} -o jsonpath={.items..metadata.name}) -c sleep -n ${from} -- curl http://httpbin.${to}:8000/ip -s -o /dev/null -w "sleep.${from} to httpbin.${to}: %{http_code}\n"; done; done
+```
+
+```
+sleep.foo to httpbin.foo: 200
+sleep.foo to httpbin.bar: 200
+sleep.bar to httpbin.foo: 200
+sleep.bar to httpbin.bar: 200
+sleep.legacy to httpbin.foo: 000
+command terminated with exit code 56
+sleep.legacy to httpbin.bar: 200
+```
+
+## Enable mTLS for single service httpbin.bar.
+
+Run this command to set another policy for only for `httpbin.bar` service. Note in this example, we don't specify namespace in metadata but as part of the commandline (`-n bar`). They should work the same.
+
+```
+cat <<EOF | istioctl create -n bar -f -
+apiVersion: "authentication.istio.io/v1alpha1"
+kind: "Policy"
+metadata:
+  name: "enable-mtls"
```

Comment:
Missing namespace in metadata.

Reply:
It's intentional :). I highlight the word in the sentence above to mention namespace is set in the command line

Reply:
I see. Does it change the config (adding namespace) after you run "-n" command?